### PR TITLE
feat: add cart page

### PR DESCRIPTION
**Pull request by anik31**
## Cart Page
I have added the cart page. Ignore the CSS & index files. Focus only on the components, pages & context + reducer files.
 
## Files to check
- [ ]  src/App.js
- [ ]  src/components/cards/ProductCard.jsx
- [ ]  src/components/navbar/Navbar.jsx
- [ ]  src/context/product-context.js
- [ ]  src/hooks/useLogin.js
- [ ]  src/pages/cart/Cart.jsx
- [ ]  src/pages/cart/CartCard.jsx
- [ ]  src/pages/cart/PriceCard.jsx
- [ ]  src/reducer/product-reducer.js
- [ ]  src/utils/serverRequests.js
 
 _PS - Ignore other files._
 
## What not to check
- [ ]  src/hooks/index.js
- [ ]  src/pages/cart/cart.css
- [ ]  src/pages/index.js
- [ ]  src/utils/index.js

https://user-images.githubusercontent.com/56336326/159156925-1f792334-6214-4e15-9986-cf7dd59c1664.mp4

### CODE REVIEW

File: src/hooks/useLogin.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export function useLogin(){
+    useEffect(()=>{
+        (async function loginHandler(){
+            try{
+                const {data} = await axios.post("/api/auth/login",{
+                    email: "johndoe@gmail.com",
+                    password: "johnDoe123"
+                })
+                localStorage.setItem("encodedToken",data.encodedToken);
+                console.log("login success");
```
Comment (foolhardy21) [inline src/hooks/useLogin.js:13]: suggestion - you may remove this console.log now.

File: src/pages/cart/Cart.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    return (
+        <>
+        <h2 className="page-title">My Cart ({state.cart.length})</h2>
+        {state.cart.length>0 &&
+        <div className="grid-container-cart">
+            <div className="card-wrapper-cart">
+                {state.cart.map(item=>{
+                    const {_id, qty, originalPrice, price} = item; 
+                    priceObj = {
+                        quantity: priceObj.quantity + Number(qty),
```
Comment (foolhardy21) [inline src/pages/cart/Cart.jsx:24]: suggestion - you can create util methods for these calculation after merging.

File: src/pages/cart/CartCard.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import "./cart.css";
+import { removeFromCart, updateItemQuantity } from "../../utils";
+import { useProducts } from "../../context/product-context";
+
+
+export function CartCard({value}){
+    const {_id ,image, title, price, originalPrice, qty, discount} = value;
```
Comment (foolhardy21) [inline src/pages/cart/CartCard.jsx:7]: suggestion -  you can destructure `value` in the argument itself.

**Review by foolhardy21 — APPROVED** [review]
page looks really good. added a few suggestions.

File: src/utils/serverRequests.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+export const getCart = async(dispatch) => {
+  try{
+    const {status, data} = await axios({
+      method: "get",
+      url: "/api/user/cart",
+      headers: {authorization: localStorage.getItem("encodedToken")}
+    });
+    console.log("Calling api get cart");
+    if(status===200){
```
Comment (BalajiHarinath) [inline src/utils/serverRequests.js:11]: remove the console.log

**Review by BalajiHarinath — APPROVED** [review]
Suggestion :- In server request.js there are console logs in many places you can remove them.

Appreciation:- The code is well structured and UI is very clean.

File: src/hooks/useLogin.js (hunk 1 above)
Comment (anik31) [reply src/hooks/useLogin.js:13]: I have added it intentionally to track the flow of the app and how many times the APIs are called. I will remove it after the project is complete.

File: src/pages/cart/Cart.jsx (hunk 2 above)
Comment (anik31) [reply src/pages/cart/Cart.jsx:24]: Okay will do this

File: src/utils/serverRequests.js (hunk 4 above)
Comment (anik31) [reply src/utils/serverRequests.js:11]: Will do it later. I have added it intentionally for now.

File: src/pages/cart/CartCard.jsx (hunk 3 above)
Comment (anik31) [reply src/pages/cart/CartCard.jsx:7]: I tried it but it doesn't work. Will find a solution later.

**Pull request merged**